### PR TITLE
WIP: Switch to upstream images as default

### PR DESCRIPTION
**Pull request by david-martin**
Implementation for #64. (downstream image https://issues.jboss.org/browse/INTLY-2596)

TODO: 

* [x] Identify and use upstream images for all containers 
* [x] Test & verify on openshift 3
* [x] Test & verify on openshift 4

To test, can use the auto built image with tag `use-upstream-images-as-default`

`quay.io/integr8ly/application-monitoring-operator:use-upstream-images-as-default`

These changes may supercede issues  #52, #58, #59, #62.

### CODE REVIEW

**Comment by david-martin** [comment]
First test on OS4 looks good. Install works. OAuth Proxy works.
Will proceed with changes to add crd flag next.

For anyone looking to try this out right now, you can apply the below change locally, the run `make cluster/install`

```diff
diff --git a/deploy/operator.yaml b/deploy/operator.yaml
index 6777658..cad2b78 100644
--- a/deploy/operator.yaml
+++ b/deploy/operator.yaml
@@ -15,7 +15,7 @@ spec:
       serviceAccountName: application-monitoring-operator
       containers:
         - name: application-monitoring-operator
-          image: quay.io/integreatly/application-monitoring-operator:0.0.17
+          image: quay.io/integreatly/application-monitoring-operator:use-upstream-images-as-default
           ports:
             - containerPort: 60000
               name: metrics
```

**Comment by david-martin** [comment]
@pb82 I'm questioning the original approach I was going to take for supporting the use of upstream and downstream (potentially protected) images.
The approach was going to be a flag in the ApplicationMonitoring CRD to say to use public or protected images (defaulting to public).
However, the idea of building in the list of protected images to this operator isn't sitting well with me.
I think that is an anti-pattern compared to any downstream/productisation we may have done in the past.

A more upstream friendly approach might be to just have it use the public images, 
but have the list of images & tags maintained in a central location (i.e. templateHelper).
Then for downstream use, we can modify the list in 1 file, and build the downstream image on RHEL from that modified source. 

How does this sound?

I realise that initially this means we'll use upstream/public images for integr8ly until such a time as we do productised builds of this operator

File: pkg/controller/applicationmonitoring/templateHelper.go [hunk 1]
```diff
@@ -89,36 +106,53 @@ type TemplateHelper struct {
...
+		BlackboxTargets:                  common.Flatten(),
+		ScrapeConfigSecretName:           ScrapeConfigSecretName,
+		ImageAlertManager:                "quay.io/openshift/origin-prometheus-alertmanager",
+		ImageTagAlertManager:             "4.2",
+		ImageOAuthProxy:                  "quay.io/openshift/origin-oauth-proxy",
+		ImageTagOAuthProxy:               "4.2",
+		ImageGrafana:                     "quay.io/openshift/origin-grafana",
+		ImageTagGrafana:                  "4.2",
+		ImageGrafanaOperator:             "quay.io/integreatly/grafana-operator",
+		ImageTagGrafanaOperator:          "v1.1.1",
```
Comment (pb82) [inline pkg/controller/applicationmonitoring/templateHelper.go]: should be v1.2.0

**Comment by david-martin** [comment]
@pb82 rebased, if you could reapprove at your convience

**Pull request merged**